### PR TITLE
[10.0] base_rest: Move from https://github.com/akretion/odoo-shopinvader

### PR DESCRIPTION
**Pull request by lmignon**
suppersedes #1 (wrong branch)
superseded by #5

### CODE REVIEW

File: base_rest/components/service.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+class BaseRestService(AbstractComponent):
+    _name = 'base.rest.service'
+
+    def _prepare_extra_log(self, func, params, secure_params, res):
+        httprequest = request.httprequest
+        headers = dict(httprequest.headers)
+        return {
+            'application': 'Rest Service',
+            'invader_url': httprequest.url,
```
Comment (HviorForgeFlow) [inline base_rest/components/service.py]: s/:space_invader:/request (maybe) ?

File: base_rest/components/service.py [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+            'invader_method': httprequest.method,
```
Comment (HviorForgeFlow) [inline base_rest/components/service.py]: s/:space_invader:/request (maybe) ?

File: base_rest/components/service.py [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+        }
+
+    def _log_call(self, func, params, secure_params, res):
+        """If you want to enjoy the advanced log install the module
+        logging_json"""
+        if request:
+            httprequest = request.httprequest
+            extra = self._prepare_extra_log(func, params, secure_params, res)
+            args = [httprequest.url, httprequest.method]
+            message = 'Invader call url %s method %s'
```
Comment (HviorForgeFlow) [inline base_rest/components/service.py]: typo: Invader

File: base_rest/controllers/main.py [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+    def collection(self):
+        return _PseudoCollection(self.collection_name, request.env)
+
+    @contextmanager
+    def work_on_component(self):
+        """
+        Return the component that implements the methods of the requested
+        service.
+        :param service_name:
+        :return: an instance of invader.service component
```
Comment (HviorForgeFlow) [inline base_rest/controllers/main.py]: s/invader.service/base.rest.service

File: base_rest/controllers/main.py [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+    @contextmanager
+    def service_component(self, service_name):
+        """
+        Return the component that implements the methods of the requested
+        service.
+        :param service_name:
+        :return: an instance of invader.service component
+        """
+        with self.work_on_component() as work:
+            service = work.component(usage=service_name)
```
Comment (HviorForgeFlow) [inline base_rest/controllers/main.py]: s/invader.service/base.rest.service

File: base_rest/controllers/main.py [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+            yield service
+
+    def _validate_method_name(self, method_name):
+        if method_name.startswith('_'):
+            _logger.error("Invader service called with an unallowed method "
```
Comment (HviorForgeFlow) [inline base_rest/controllers/main.py]: Typo: Invader, Rest API maybe?

**Review by HviorForgeFlow — CHANGES_REQUESTED** [review]
Just a few typo comments. Functional review. :+1: It seems is working as expected.

**Comment by lmignon** [comment]
@hveficent Thank you the review. I'll change the code to remove the last remaining references to 'shopinvader' ;-)
 If you plan to use this addon you have to know that the API is not yet stabilized. It will change in a near future.  see #2 Theses API changes will help to properly put in place a swagger frontend to document (and manually test) the services #5

**Review by sebastienbeau — APPROVED** [review]
Thank for this awesome project. When it will be merge I will remove the code from shopinvader project.

**Comment by lmignon** [comment]
@sebastienbeau I discovered some bugs yesterday.
* c8bb99d14199a33c4b7a7d112a263e36d92fbb8f
* 768d12ec053649349301ea843864825bad4e975b
* e7430e722131f79d8fc044eecb1291275add917c
I could move these commits into the 10.0-add_base_rest branch but I wonder if it would be better to create the first 10.0 release from #5

**Pull request closed without merging**